app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java

```diff
@@ -945,4 +949,126 @@ public void showEventWasDeleted() {
+            }
+
+            @Override
+            public void mapRequest(@NotNull String coordinateFieldUid, @NotNull String featureType, @org.jetbrains.annotations.Nullable String initialCoordinates) {
+                startActivityForResult(
+                        MapSelectorActivity.Companion.create(EventInitialActivity.this,
+                                FeatureType.valueOfFeatureType(featureType),
+                                initialCoordinates),
+                        RQ_MAP_LOCATION_VIEW);
+            }
+
+            @Override
+            public void onItemAction(@NotNull RowAction action) {
+                presenter.setChangingCoordinates(true);
+                setNewGeometry(action.getValue());
+                setGeometryModel((CoordinateViewModel) geometryModel.withValue(action.getValue()));
+            }
+        });
+    }
+
+    @Override
+    public void setNewGeometry(String value) {
+        if(value != null) {
+            this.newGeometry = Geometry.builder()
+                    .coordinates(value)
+                    .type(programStage.featureType())
+                    .build();
+        }else{
+            this.newGeometry = null;
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        if (resultCode == RESULT_OK && requestCode == RQ_MAP_LOCATION_VIEW && data != null && data.getExtras() != null) {
+            FeatureType locationType = FeatureType.valueOf(data.getStringExtra(MapSelectorActivity.LOCATION_TYPE_EXTRA));
+            String dataExtra = data.getStringExtra(MapSelectorActivity.DATA_EXTRA);
+            Geometry geometry = null;
+            switch (locationType) {
+                case POINT:
+                    geometry = GeometryHelper.createPointGeometry(
+                            new Gson().fromJson(
+                                    dataExtra,
+                                    new TypeToken<List<Double>>() {
+                                    }.getType()
+                            ));
+                    break;
+                case POLYGON:
+                    geometry = GeometryHelper.createPolygonGeometry(
+                            new Gson().fromJson(
+                                    dataExtra,
+                                    new TypeToken<List<List<List<Double>>>>() {
+                                    }.getType()
+                            ));
+                    break;
+                case MULTI_POLYGON:
+                    geometry = GeometryHelper.createMultiPolygonGeometry(
+                            new Gson().fromJson(
+                                    dataExtra,
+                                    new TypeToken<List<List<List<List<Double>>>>>() {
+                                    }.getType()
+                            ));
+                    break;
+                case NONE:
+                    break;
+            }
```

Comment:
This is also repeated in other activities should we have a controller or something?